rllab/envs/dmcontrol_env.py

```diff
@@ -0,0 +1,130 @@
+import pygame
+import numpy as np
+from PIL import Image
+from dm_control import suite
+from dm_control.rl.environment import StepType
+from dm_control.suite.wrappers import pixels
+from rllab.envs.base import Env, Step
+from rllab.core.serializable import Serializable
+from rllab.spaces.box import Box
+from rllab.spaces.discrete import Discrete
+
+
+'''
+This environment will use dm_control toolkit to train and simulate your model.
+Domain name and task name used for dm_control is showed below: (domain name/task name)
+
+ball_in_cup/catch
+cartpole/swingup_sparse
+cartpole/balance_sparse
+cartpole/swingup
+cartpole/balance
+cheetah/run
+finger/turn_hard
+finger/spin
+finger/turn_easy
+fish/upright
+fish/swim
+hopper/stand
+hopper/hop
+humanoid/stand
+humanoid/run
+humanoid/walk
+manipulator/bring_ball
+pendulum/swingup
+point_mass/easy
+reacher/hard
+reacher/easy
+swimmer/swimmer6
+swimmer/swimmer15
+walker/stand
+walker/run
+walker/walk
+'''
+
+
+def flat_shape(observation):
+    dim = 0
+    for key, value in observation.items():
+        dim += int(np.prod(value.shape))
+    return dim
+
+def flat_observation(observation):
+    new_observation=[]
+    for key, value in observation.items():
+        new_observation += value.flatten().tolist()
+    observation = np.array(new_observation)
+    return observation
+
+
+class DMControlEnv(Env, Serializable):
+    def __init__(self, domain_name, task_name = "swingup", 
+            display = True, width = 320, height = 240, pixel_key = 'rbg', pixels_only = False, *arg, **kwargs):
+        Serializable.quick_init(self, locals())
+        pygame.init()
+
+        #create pygame window
+        self.display_window = pygame.display.set_mode((width,height))
+
+        #create dm control env
+        self.env = suite.load(domain_name=domain_name, task_name=task_name)
+        self.wrapped = pixels.Wrapper(self.env,
+                      observation_key=pixel_key,
+                      pixels_only=pixels_only,
+                      render_kwargs={'width': width, 'height': height})
+
+        self.pixel_key = pixel_key
+        self._display = display
+        self._observation_spec = self.wrapped.observation_spec()
```

Comment:
no need to store these when they can be accessed through `self.wrapped`. It makes the code easier to maintain.

Reply:
Done